src/wasm-lib/kcl/src/ast/types.rs

```diff
@@ -2885,6 +2898,7 @@ impl BinaryOperator {
         match self {
             Self::Add | Self::Sub | Self::Mul | Self::Div | Self::Mod => Associativity::Left,
             Self::Pow => Associativity::Right,
+            Self::Eq | Self::Neq => Associativity::Left, // I don't know if this is correct
```

Comment:
tl;dr: looks good 😄 Read on for the gory details.

I'm pretty sure that comparison operators are usually non-associative. If they were left associative, this:

```
a == b == c
```

would parse like

```
(a == b) == c
```

and if the variables are numbers, the result of comparison is a boolean, meaning the second comparison is between a boolean and a number.

In a statically typed language, this would be a type error. But with dynamic typing, it could parse and fail with a runtime error. I find it a little strange that we can't compare other types (e.g. string with string, boolean with boolean). But this works.

I think it's important that the second comparison, i.e. comparisons between non-numbers, fails, not simply return false. If we returned false, then the expression `a == b == c` could always be simplified to `false`, and that's certainly a bug in the user's program.

The fancy way of doing it that's more ergonomic, but also more work to implement, is to make it like pipelines, where it bundles `a == b == c` into a single expression that desugars to `a == b && b == c`.

To summarize, I think we need to pick one of these:

1. left-associative, second comparison is a runtime error (this PR currently does this)
2. non-associative (i.e. parse error)
3. parse the whole chain and desugar to `a == b && b == c`

Reply:
I prefer non-associative. Hmm. I've got some work to do here.

Reply:
@jtran so right now `3==3==3` and `(3==3)==3` won't compile, so it should be OK. I added a test to prove that.